tests/config/test_defaults.py

```diff
@@ -41,93 +42,234 @@ def test_create_defaults_without_permission() -> None:
+    expect: list[str],
+    tmp_path: Path,
 ) -> None:
     """Test getting a list of strings from defaults.ini using a custom delimiter."""
-    content = """
-    [test.list]
-    default =
-        github.com
-        comma_ended,
-        space string
-        space string
-    empty =
-    one_line = github.com comma_ended, space string space string
-    commas_string = ,github.com, gitlab.com, space string, space string
-    """
-    custom_defaults = ConfigParser()
-    custom_defaults.read_string(content)
+    user_config_path = os.path.join(tmp_path, "config.ini")
+    with open(user_config_path, "w", encoding="utf-8") as user_config_file:
+        user_config_file.write(user_config_input)
+    load_defaults(user_config_path)
 
-    results = custom_defaults.get_list(section, item, delimiter=delimiter, strip=strip, duplicated_ok=duplicated_ok)
-    results.sort()
+    results = defaults.get_list(section="test.list", option="list", delimiter=delimiter, strip=strip)
     assert results == expect
 
 
 @pytest.mark.parametrize(
-    ("section", "item", "strip", "duplicated_ok", "fallback", "expect"),
+    ("user_config_input", "expect"),
     [
-        ("test.list", "default", True, False, [], ["comma_ended,", "github.com", "space string"]),
-        ("test.list", "default", True, True, [], ["comma_ended,", "github.com", "space string", "space string"]),
-        ("test.list", "empty", False, True, [], [""]),
-        ("test.list", "empty", True, True, [], []),
-        # Test for an item that does not exist in defaults.ini
-        ("test.list", "item_not_exist", True, True, [], []),
-        # Test value for fallback. The fallback value must be returned as is and shouldn't be modified by the method.
-        ("test.list", "item_not_exist", True, True, ["", "fallback_val"], ["", "fallback_val"]),
+        (
+            """
+            [test.list]
+            list = ,github.com, gitlab.com, space string, space string
+            """,
+            [",github.com, gitlab.com, space string, space string"],
+        ),
+        (
+            """
+            [test.list]
+            list =
+                github.com
+                comma_ended,
+                space string
+                foo bar
+                foo bar
+                space string
+            """,
+            ["github.com", "comma_ended,", "space string", "foo bar"],
+        ),
+        (
+            """
+            [test.list]
+            list =
+            """,
+            [],
+        ),
     ],
 )
-def test_get_str_list_with_default_delimiter(
-    section: str, item: str, strip: bool, duplicated_ok: bool, fallback: list[str], expect: list[str]
+def test_get_str_list_default(
+    user_config_input: str,
+    expect: list[str],
+    tmp_path: Path,
 ) -> None:
-    """Test getting a list of strings from defaults.ini using the default delimiter."""
+    """Test default behavior of getting a list of strings from an option in defaults.ini."""
```

Comment:
Can you please be more specific what the default behavior is?

Reply:
Addressed in https://github.com/oracle/macaron/pull/660/commits/87cd0f9b38681b1b741bf6a2c43a1f271d036247